src/plugins/data/common/search/aggs/buckets/create_filter/date_range.ts

```diff
@@ -6,15 +6,16 @@
  * Side Public License, v 1.
  */
 
-import moment from 'moment';
+import moment from 'moment-timezone';
 import { buildRangeFilter, RangeFilterParams } from '@kbn/es-query';
 import { DateRange } from '../../../expressions';
 import { IBucketAggConfig } from '../bucket_agg_type';
 
 export const createFilterDateRange = (agg: IBucketAggConfig, { from, to }: DateRange) => {
   const filter: RangeFilterParams = {};
-  if (from) filter.gte = moment(from).toISOString();
-  if (to) filter.lt = moment(to).toISOString();
+  if (from) filter.gte = moment.tz(from, agg.aggConfigs.timeZone).toISOString();
```

Comment:
Note: Shouldn't matter in this case as no date calculation is happening, but can't hurt to do it right